src/ui_iteminfo.cpp

```diff
@@ -0,0 +1,195 @@
+                                this_color = c_light_gray;
+                            } else if( iVal > kVal ) {
+                                if( i.bLowerIsBetter ) {
+                                    this_color = c_light_red;
+                                } else {
+                                    this_color = c_light_green;
+                                }
+                            } else if( iVal < kVal ) {
+                                if( i.bLowerIsBetter ) {
+                                    this_color = c_light_green;
+                                } else {
+                                    this_color = c_light_red;
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
+                formatted_string += colorize( i.sValue, this_color );
+            }
+
+            if( pos != std::string::npos ) {
+                formatted_string += sFmt.substr( pos + 5 );
+            }
+
+            if( !formatted_string.empty() ) {
+                if( same_line ) {
+                    ImGui::SameLine( 0, 0 );
+                }
+                cataimgui::draw_colored_text( formatted_string, base_color, wrap_width );
+            }
+            same_line = !i.bNewLine;
+        }
+    }
+
+    set_scroll( s );
+}
+
+iteminfo_window::iteminfo_window( item_info_data &info, point pos, int width, int height,
+                                  ImGuiWindowFlags flags ) :
+    cataimgui::window( _( "Item info" ), flags ),
+    data( info ), pos( pos ), width( width ), height( height )
+{
+    ctxt = input_context( "default" );
+    if( info.handle_scrolling ) {
+        ctxt.register_action( "PAGE_UP" );
+        ctxt.register_action( "PAGE_DOWN" );
+        if( info.arrow_scrolling ) {
+            ctxt.register_action( "UP" );
+            ctxt.register_action( "DOWN" );
+        }
+    }
+    ctxt.register_action( "CONFIRM" );
+    ctxt.register_action( "QUIT" );
+    if( info.any_input ) {
+        ctxt.register_action( "ANY_INPUT" );
+    }
+
+    ctxt.set_timeout( 10 );
+}
+
+cataimgui::bounds iteminfo_window::get_bounds()
+{
+    return { pos.x < 0 ? -1.f : static_cast<float>( str_width_to_pixels( pos.x ) ),
+             pos.y < 0 ? -1.f : static_cast<float>( str_height_to_pixels( pos.y ) ),
+             static_cast<float>( str_width_to_pixels( width ) ),
+             static_cast<float>( str_height_to_pixels( height ) ) };
+}
+
+void iteminfo_window::draw_controls()
+{
+    draw_item_info_imgui( *this, data, width, s );
+}
+
+void iteminfo_window::execute()
+{
+    if( data.without_getch ) {
+        ui_manager::redraw();
+        return;
+    }
+
+    while( true ) {
+        ui_manager::redraw();
+        std::string action = ctxt.handle_input();
+
+        if( data.handle_scrolling && data.arrow_scrolling && action == "UP" ) {
+            s = scroll::line_up;
```

Comment:
I have thoughts about this, but there is no need to delay this patch because of them.

Reply:
Tbh I'm not sure myself if the `arrow_scrolling` is needed as a separate thing. At least for the ones I switched over to imgui here (and the V menu rewrite) it's not necessary. But at least there could be cases where up/down is already blocked by other functionality, so I did it like that for now.

Feel free to also add other thoughts you had anyway, even if we decide to not put them in this PR.

Reply:
Oh, the `arrow_scrolling` flag is fine. I’m referring to the whole way we compute the scroll amount and eventually communicate it to ImGui. I filed #76007 just now with the whole spiel.

Reply:
Oh yeah, one of the reasons I set it up the way I did is so it could fairly easily be ripped out, generalized and extended or replaced by something else.